pages/admin/users.tsx

```diff
@@ -38,23 +46,63 @@ export default function UserPage() {
     super_admin: true,
   });
 
-  async function fetchAllUsers() {
+  async function fetchInitData() {
     setLoading(true);
+    setNextRegistrationStatus(RegistrationState.UNINITIALIZED);
     if (!user) return;
 
-    const { data } = await RequestHelper.get<UserIdentifier[]>('/api/users', {
-      headers: {
-        Authorization: user.token,
-      },
-    });
+    const usersData = (
+      await RequestHelper.get<UserIdentifier[]>('/api/users', {
+        headers: {
+          Authorization: user.token,
+        },
+      })
+    )['data'];
+
+    const allowRegistrationState = (
+      await RequestHelper.get<{ allowRegistrations: boolean }>('/api/registrations/status', {
+        headers: {
+          Authorization: user.token,
+        },
+      })
+    )['data'];
+
+    const hackersStatus = (
+      await RequestHelper.get<HackerStatus[]>(`/api/acceptreject?adminId=${user.id}`, {
+        headers: {
+          Authorization: user.token,
+        },
+      })
+    )['data'];
+
+    setRegistrationStatus(
+      allowRegistrationState.allowRegistrations ? RegistrationState.OPEN : RegistrationState.CLOSED,
+    );
+    const accepted = [],
+      rejected = [];
 
-    setUsers(data);
-    setFilteredUsers([...data]);
+    for (const hackerStatus of hackersStatus) {
+      if (hackerStatus.status === 'Accepted') accepted.push(hackerStatus.hackerId);
+      else if (hackerStatus.status === 'Rejected') rejected.push(hackerStatus.hackerId);
+      else throw 'Unknown status';
+    }
+
+    for (const userData of usersData) {
+      userData.status = accepted.includes(userData.id)
+        ? 'Accepted'
+        : rejected.includes(userData.id)
+        ? 'Rejected'
+        : 'Waiting';
+      userData.selected = false;
+    }
```

Comment:
Will need to re-implement this if we have like >= 1000 hackers because this is $O(n^2)$.